tests/suites/test_suite_x509write.function

```diff
@@ -340,6 +341,25 @@ void x509_crt_check( char *subject_key_file, char *subject_pwd,
             TEST_ASSERT( mbedtls_x509write_crt_set_key_usage( &crt, key_usage ) == 0 );
         if( set_cert_type != 0 )
             TEST_ASSERT( mbedtls_x509write_crt_set_ns_cert_type( &crt, cert_type ) == 0 );
+        if( strcmp( ext_key_usage, "NULL" ) != 0 )
+        {
+            mbedtls_asn1_sequence exts[2];
+            memset(exts, 0, sizeof(exts));
+#define SET_OID(x, oid) \
+    do { x.len = MBEDTLS_OID_SIZE(oid); x.p = (unsigned char*)oid; \
+         x.tag = MBEDTLS_ASN1_OID; } while( 0 )
```

Comment:
The style guide says to put every single brace on its own line:
```
do
{
    …
}
while( 0 )
```
I don't like separating the closing brace of a do-while from the `while`, because then when you're looking at the `while` you can't easily see that it's part of a do-while. But that is what the style guide says and what we've generally been doing.

Reply:
@gilles-peskine-arm Specifically regarding this, we can and do "approve" exceptions to the style guide when it makes the code easier to read, particularly when it's something that's done rarely. I don't see any reason why we couldn't consider "closing brace of do-while" as one of these - it's nearly always used in macros, and macros are handled differently by the style guide anyway

Reply:
Oh, I'm personally all for that. I can see the rationale for lining up `{` with `}`, and thus having a line break between `do` and `{` (even though my personal preference is for Egyptian braces). But I've never understood why `while` would be on its own line.

I don't think anybody currently on the Mbed TLS maintainer team would object.

Reply:
too late!